sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -17,6 +19,13 @@ def __init__(self, endpoint, headers, retry_policy):
         self._endpoint = endpoint
         self._headers = headers
         self._retry_policy = retry_policy
+        if self._endpoint.auth == constants.AUTH_KERBEROS:
+            self._auth = HTTPKerberosAuth(mutual_authentication=REQUIRED)
+        elif self._endpoint.auth == constants.AUTH_BASIC:
+            self._auth = (self._endpoint.username, self._endpoint.password)
+        elif self._endpoint.auth != constants.NO_AUTH:
+            raise ValueError(u"Unsupported auth."+self._endpoint.auth)
```

Comment:
This is still ValueError when I asked to change it to Bad Config exception at https://github.com/jupyter-incubator/sparkmagic/blob/master/sparkmagic/sparkmagic/livyclientlib/exceptions.py#L41